File: backend/src/main/scala/microservices/train/objects/TrainListPlannerResponse.scala
package com.typesafe.travel.train.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TrainListPlannerResponse(trains: List[TrainPlannerResponse])
object TrainListPlannerResponse:
  given sourceEncoder: Encoder[TrainListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[TrainListPlannerResponse] = deriveDecoder

File: frontend/src/microservices/train/objects/TrainListPlannerResponse.ts
// 本文件定义 train 模块的 `TrainListPlannerResponse`，用于表示火车列表结果并提供 JSON 编解码。

import type { TrainPlannerResponse } from './TrainPlannerResponse'

export type TrainListPlannerResponse = {
  trains: TrainPlannerResponse[]
}
export const trainListPlannerResponseFromJson = (json: string): TrainListPlannerResponse =>
  JSON.parse(json) as TrainListPlannerResponse

export const trainListPlannerResponseToJson = (value: TrainListPlannerResponse): string =>
  JSON.stringify(value)
